src/languages/css-language.js

```diff
@@ -125,6 +154,58 @@ export class CSSLanguage {
 		}
 	}
 
+	/**
+	 * Normalizes the language options so they can be serialized.
+	 * @param {CSSLanguageOptions} languageOptions The language options to normalize.
+	 * @returns {CSSLanguageOptions} The normalized language options.
+	 */
+	normalizeLanguageOptions(languageOptions) {
+		// if there's no custom syntax then no changes are necessary
+		if (!languageOptions?.customSyntax) {
+			return languageOptions;
+		}
+		// Shallow copy
+		const clone = { ...languageOptions };
+
+		Object.defineProperty(clone, "toJSON", {
+			value() {
+				// another shallow copy
+				const result = { ...this };
+
+				// if there's no custom syntax, no changes are necessary
+				if (!this.customSyntax) {
+					return result;
+				}
+
+				result.customSyntax = { ...result.customSyntax };
+
+				if (result.customSyntax.node) {
+					result.customSyntax.node = replaceFunctions(
+						result.customSyntax.node,
+					);
+				}
+
+				if (result.customSyntax.scope) {
+					result.customSyntax.scope = replaceFunctions(
+						result.customSyntax.scope,
+					);
+				}
+
+				if (result.customSyntax.atrule) {
+					result.customSyntax.atrule = replaceFunctions(
+						result.customSyntax.atrule,
+					);
+				}
```

Comment:
It looks like this could be done in one line:

```js
				result.customSyntax = replaceFunctions(result.customSyntax);
```

Somehow GitHub isn't letting me add suggestions over multiple lines 🤷